server/src/main/java/org/opensearch/index/mapper/DerivedFieldType.java

```diff
@@ -134,6 +146,11 @@ public DerivedFieldValueFetcher valueFetcher(QueryShardContext context, SearchLo
         );
     }
 
+    @Override
+    public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, Supplier<SearchLookup> searchLookup) {
+        return typeFieldMapper.mappedFieldType.fielddataBuilder(fullyQualifiedIndexName, searchLookup);
```

Comment:
just curios about where `fieldDataBuilder()` is used?

Reply:
This is used to build FieldContext [here](https://github.com/opensearch-project/OpenSearch/blob/5c150d5716393147dbe97102da128f1b9160db7b/server/src/main/java/org/opensearch/search/aggregations/support/ValuesSourceConfig.java#L176)  that is then used to determine the `ValueSourceType`.  If left unimplemented this will throw as not supported from `MappedFieldType`. 

Its then set to null in the config so that we call getScript [here](https://github.com/opensearch-project/OpenSearch/blob/5c150d5716393147dbe97102da128f1b9160db7b/server/src/main/java/org/opensearch/search/aggregations/support/ValuesSourceConfig.java#L349).

For most agg types this isn't really required as the defaultValueSourceType works fine.  However for some, matrix stats that use `ArrayValuesSourceAggregationBuilder` the default is not always valid.